### PR TITLE
NetBSD support.

### PR DESCRIPTION
**Pull request by alarixnia**

### CODE REVIEW

File: Source/Core/Common/MemoryUtil.cpp [hunk 1]
```diff
@@ -156,15 +156,15 @@ size_t MemPhysical()
...
   int mib[2];
   size_t physical_memory;
   mib[0] = CTL_HW;
 #ifdef __APPLE__
   mib[1] = HW_MEMSIZE;
 #elif defined __FreeBSD__
   mib[1] = HW_REALMEM;
-#elif defined __OpenBSD__
+#elif defined __NetBSD__ || defined __OpenBSD__
   mib[1] = HW_PHYSMEM;
```
Comment (krytarowski) [inline Source/Core/Common/MemoryUtil.cpp]: NetBSD has to use HW_PHYSMEM64.

File: Source/Core/Common/Thread.cpp [hunk 2]
```diff
@@ -135,6 +137,8 @@ void SetCurrentThreadName(const char* szThreadName)
   pthread_setname_np(szThreadName);
 #elif defined __FreeBSD__ || defined __OpenBSD__
   pthread_set_name_np(pthread_self(), szThreadName);
+#elif defined __NetBSD__
+  pthread_setname_np(pthread_self(), szThreadName, NULL);
```
Comment (krytarowski) [inline Source/Core/Common/Thread.cpp]: `pthread_setname_np(pthread_self(), "%s", (void *)szThreadName);`

File: CMakeLists.txt [hunk 3]
```diff
@@ -321,12 +321,18 @@ endif()
...
 # changes, so just setting them in the project scope via
 # include_directories and link_directories is not sufficient
-if(CMAKE_SYSTEM_NAME MATCHES "FreeBSD|NetBSD")
+if(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
   set(CMAKE_PREFIX_PATH "${CMAKE_PREFIX_PATH};/usr/local")
   set(CMAKE_REQUIRED_INCLUDES "/usr/local/include")
   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -L/usr/local/lib")
 endif()
 
+if(CMAKE_SYSTEM_NAME MATCHES "NetBSD")
```
[the remark this answers is not in the document]
Comment (ensra) [reply CMakeLists.txt]: This part can probably be removed, tbh.

File: Source/Core/Common/MemoryUtil.cpp [hunk 4]
```diff
@@ -156,14 +156,16 @@ size_t MemPhysical()
   memInfo.dwLength = sizeof(MEMORYSTATUSEX);
   GlobalMemoryStatusEx(&memInfo);
   return memInfo.ullTotalPhys;
-#elif defined __APPLE__ || defined __FreeBSD__ || defined __OpenBSD__
+#elif defined __APPLE__ || defined __FreeBSD__ || defined __NetBSD__ || defined __OpenBSD__
   int mib[2];
   size_t physical_memory;
```
Comment (krytarowski) [inline Source/Core/Common/MemoryUtil.cpp]: NetBSD needs to use `uint64_t` at least for endian purity.

**Comment by MerryMage** [comment]
Depends on SFML/SFML#1535?

**Comment by jordan-woyak** [comment]
Is this PR complete? Can we get a rebase?

**Comment by Dentomologist** [comment]
I believe this is superseded by https://github.com/dolphin-emu/dolphin/pull/9309.

**Comment by jordan-woyak** [comment]
Stale PR. Closing in favor of already merged PR #9309.

**Pull request closed without merging**